frontend/src/components/UserUpdateRoles.vue

```diff
@@ -1,210 +1,384 @@
-                      {{clientRoles[roleArrayPosition(col, item)].name}}
-                      <span v-show="clientRoles[roleArrayPosition(col, item)].description" class="tooltiptext"> {{ clientRoles[roleArrayPosition(col, item)].description }} </span>
-                    </span>
-                  </v-checkbox>
-                </span>
-              </v-col>
-            </v-row>
-          </v-col>
-          <v-col class="col-4" no-gutters>
-            <v-row no-gutters>
-              <v-col class="col-12">
-                <label>
-                  Effective Roles
-                  <v-tooltip right>
-                    <template v-slot:activator="{ on }">
-                      <v-icon v-on="on" small>mdi-help-circle</v-icon>
-                    </template>
-                    <span>
-                      Effective roles represent all roles assigned to a user for this client.
-                      <br />This may include roles provided by group membership which cannot be directly removed.
-                    </span>
-                  </v-tooltip>
-                </label>
-              </v-col>
-              <v-col class="col-6">
-                <v-checkbox
-                    class="roles-checkbox"
-                    hide-details="auto"
-                    disabled
-                    readonly
-                    v-for="role in effectiveClientRoles"
-                    v-model="effectiveClientRoles"
-                    :value="role"
-                    :key="role.name"
+            <!-- pop up to add something -->
+            <v-card class="popup">
+              <!-- pop up header -->
+              <v-row class="header">
+                <label for="select-client">Application</label>
+                <v-icon @click="close()">mdi-close</v-icon>
+              </v-row>
+              <!-- pop up content: -->
+              <!-- client selector -->
+              <v-row>
+                <v-col class="col-7">
+                  <v-autocomplete
+                    id="select-client"
+                    outlined
+                    dense
+                    :items="isEdit ? [selectedClient] : clientWithoutRoles"
+                    item-text="clientId"
+                    return-object
+                    placeholder="Select an Application"
+                    v-model="selectedClient"
+                    @change="getUserClientRoles()"
+                    :disabled="isEdit"
+                  ></v-autocomplete>
+                </v-col>
+              </v-row>
+              <v-skeleton-loader
+                ref="roleSkeleton"
+                v-show="
+                  selectedClient && (!clientRoles || clientRoles.length == 0)
+                "
+                type="article, button"
+              ></v-skeleton-loader>
+
+              <!-- role list -->
+              <div v-if="selectedClient">
+                <div
+                  width="inherit"
```

Comment:
My IDE reports that `width` is not a valid `div` attribute. I think it's probably right, reading [StackOverflow](https://stackoverflow.com/questions/2185519/what-are-valid-attributes-for-the-div-element-in-html).

`Attribute width is not allowed here`